### PR TITLE
[labeling] Fix minimum feature size settings not taking merged lines into account

### PR DESCRIPTION
**Pull request by nirvn**
## Description

The title says it all. The PR fixes the labelling engine's minimum feature size when the merged connected lines is toggled on. The long story short, since the merging of connected lines takes place within pal, we should only apply the minimum feature size requirement against merged lines at that stage.

### CODE REVIEW

**Comment by roya0045** [comment]
Would that fixe cases where line merging doesn't seem to connect all the lines when a small segment is present? Or would that be a distinct issue in the merger itself.

**Comment by nirvn** [comment]
@roya0045 , if you have a minimum feature size that filters out these small segments, yes.

**Comment by roya0045** [comment]
@nirvn Excellent! Thank you for the fix.

File: src/core/pal/layer.cpp
Comment (nyalldawson) [inline src/core/pal/layer.cpp:345]: Looking at this again, it's actually leaking here because we are only removing the pointers from the list, but not actually deleting them. 

```suggestion
    if ( part->feature()->minimumSize() != 0.0 && part->length() < part->feature()->minimumSize() )
    {
        delete part;
        return true;
    }
    else
    {
      return false;
    }
```

Comment (nyalldawson) [reply src/core/pal/layer.cpp:345]: (after we merge this i'll go back and see if we can make mFeatureParts a list of unique_ptrs)

**Review by nyalldawson — COMMENTED** [review]
Any chance of a test too?

**Comment by nirvn** [comment]
@nyalldawson , review addressed, test case added.

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**
